api/src/main/java/javax/enterprise/inject/spi/builder/InjectionPointConfigurator.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.spi.builder;
+
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.spi.AnnotatedField;
+import javax.enterprise.inject.spi.AnnotatedParameter;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.inject.spi.ProcessInjectionPoint;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+/**
+ * This API is an helper to configure a new {@link InjectionPoint} instance.
+ * CDI container must provides an implementation of this interface.
+ *
+ * InjectionPointConfigurator is not reusable.
+ *
+ * This builder is not thread safe and shall not be used concurrently.
+ *
+ * @see ProcessInjectionPoint#configureInjectionPoint()
+ * @author Antoine Sabot-Durand
+ * @since 2.0
+ */
+public interface InjectionPointConfigurator {
```

Comment:
Well, the InjectionPoint should be initialized by one of the the read method (an other IP or a field or  a parameter). `member()` and `annotated()` will be copied from there. Now, do you see some use case where it would be useful to override them ?